client/src/pages/Login.tsx

```diff
@@ -12,15 +12,21 @@ const login = async ({
   email: string;
   password: string;
 }) => {
-  const res = await fetch(getApiUrl("/auth/login"), {
+  // Interviewee Task - send appropriate headers for basic auth
+  const loginRequestOptions = {
     method: "POST",
-    // Interviewee Task - add basic auth headers for login
-  });
+    headers: {
+      "Authorization": `Basic ${btoa(`${email}:${password}`)}`,
+    },
+  };
+  console.dir(loginRequestOptions);
+  const res = await fetch(getApiUrl("/auth/login"), loginRequestOptions);
+  console.dir(res);
   return res.json();
 };
 
 export const Login = () => {
-  const { loggedIn } = useAuthContext();
+  const { loggedIn, setLoggedInState } = useAuthContext();
```

Comment:
Agreed you would need a way to track logging in and the state.
[React Query [useMutation]](https://tanstack.com/query/v4/docs/framework/react/reference/useMutation) actually tracks this internally and exposes, could leverage to tie in with the `AuthContext`.

There being a many new frameworks within in, perfectly fine the way you are tracking, as a PR from a teammate would help in understanding React Query further 😄